graph_model/trie_model/dataset.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+import string
+
+class Dataset:
+        
+    def __init__(self):
+        self.df = self.read_data()
+        self.preprocess()
+
+
+    def read_data(self):
+        df = pd.read_csv('./temp/data/item_master.csv.gz', compression='gzip')
+        return df['PRODUCT_NAME'].values
+
+
+    def preprocess(self):
```

Comment:
```suggestion
    def preprocess_title(self, item_title:str) -> str:
```
Let's have this function to preprocess titles one by one for simplicity.